### PR TITLE
[fix]:[CCM-10575]: NodePoolRecommendation fix with default value incase of timeout- 809xx hfix-2

### PR DESCRIPTION
**Pull request by abhinavsingh-ccm**
## Describe your changes
Logic has been updated to fetch totalResource for that specific day at which job is running. No more calculation of `minStartTime`.

Sample Query:

```java
select
  max("alias_20404458"."sumcpu") as "sumcpu",
  max("alias_20404458"."summemory") as "summemory",
  max("alias_20404458"."maxcpu") as "maxcpu",
  max("alias_20404458"."maxmemory") as "maxmemory"
from (
  select
    sum("alias_99922606"."cpu") as "sumcpu",
    sum("alias_99922606"."memory") as "summemory",
    max("alias_99922606"."maxcpu") as "maxcpu",
    max("alias_99922606"."maxmemory") as "maxmemory"
  from (
    select
      "public"."time_bucket"(
        "bucket_width" := cast('+0-0 +0 00:20:00.000000000' as interval),
        "ts" := cast("public"."kubernetes_utilization_data"."starttime" as timestamp with time zone)
      ) as "time",
      greatest(
        max("public"."kubernetes_utilization_data"."cpu"),
        max("alias_81236478"."cpurequest")
      ) as "cpu",
      greatest(
        max("public"."kubernetes_utilization_data"."memory"),
        max("alias_81236478"."memoryrequest")
      ) as "memory",
      greatest(
        max("public"."kubernetes_utilization_data"."maxcpu"),
        max("alias_81236478"."cpurequest")
      ) as "maxcpu",
      greatest(
        max("public"."kubernetes_utilization_data"."maxmemory"),
        max("alias_81236478"."memoryrequest")
      ) as "maxmemory"
    from "public"."kubernetes_utilization_data"
      join (
        select
          ("public"."pod_info"."namespace" || '/' || "public"."pod_info"."name") as "actualinstanceid",
          "public"."pod_info"."cpurequest",
          "public"."pod_info"."memoryrequest"
        from "public"."pod_info"
          join "public"."node_info"
            on (
              "public"."node_info"."instanceid" = "public"."pod_info"."parentnodeid"
              and "public"."node_info"."accountid" = 'kmpySmUISimoRrJL6xxxxx'
              and "public"."node_info"."clusterid" = '60fd050d760161a7540xxxx'
              and "public"."pod_info"."accountid" = 'kmpySmUISimoRrJL6xxxxx'
              and "public"."pod_info"."clusterid" = '60fd050d760161a7540xxxx'
              and "public"."node_info"."nodepoolname" = 'general-preemptible'
              and "public"."is_alive"(
                "instancestarttime" := "public"."node_info"."starttime",
                "instancestoptime" := "public"."node_info"."stoptime",
                "jobstarttime" := timestamp with time zone '2023-10-02 00:00:00+00:00',
                "jobstoptime" := timestamp with time zone '2023-10-03 00:00:00+00:00'
              ) = true
              and "public"."pod_info"."namespace" <> 'kube-system'
              and "public"."is_alive"(
                "instancestarttime" := "public"."pod_info"."starttime",
                "instancestoptime" := "public"."pod_info"."stoptime",
                "jobstarttime" := timestamp with time zone '2023-10-02 00:00:00+00:00',
                "jobstoptime" := timestamp with time zone '2023-10-03 00:00:00+00:00'
              ) = true
              and "public"."pod_info"."starttime" >= timestamp with time zone '2023-10-02 00:00:00+00:00'
              and "public"."pod_info"."starttime" < timestamp with time zone '2023-10-03 00:00:00+00:00'
              and "public"."node_info"."starttime" >= timestamp with time zone '2022-10-02 00:00:00+00:00'
            )
      ) as "alias_81236478"
        on (
          "public"."kubernetes_utilization_data"."actualinstanceid" = "alias_81236478"."actualinstanceid"
          and "public"."kubernetes_utilization_data"."accountid" = 'kmpySmUISimoRrJL6xxxxx'
          and "public"."kubernetes_utilization_data"."clusterid" = '60fd050d760161a7540xxxx'
          and "public"."is_alive"(
            "instancestarttime" := "public"."kubernetes_utilization_data"."starttime",
            "instancestoptime" := "public"."kubernetes_utilization_data"."endtime",
            "jobstarttime" := timestamp with time zone '2023-10-02 00:00:00+00:00',
            "jobstoptime" := timestamp with time zone '2023-10-03 00:00:00+00:00'
          ) = true
          and "public"."kubernetes_utilization_data"."starttime" >= timestamp with time zone '2023-10-02 00:00:00+00:00'
          and "public"."kubernetes_utilization_data"."starttime" < timestamp with time zone '2023-10-03 00:00:00+00:00'
        )
    group by
      "time",
      "public"."kubernetes_utilization_data"."actualinstanceid"
  ) as "alias_99922606"
  group by "alias_99922606"."time"
) as "alias_20404458"
``` 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53707)

### CODE REVIEW

**Comment by abhinavsingh-ccm** [comment]
trigger uts

File: 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java [hunk 1]
```diff
@@ -390,7 +317,8 @@ private Table<? extends Record> getPodRequest(
             POD_INFO.NAMESPACE.notEqual(kube_system_namespace),
             TimescaleUtils.isAlive(
                 POD_INFO.STARTTIME, POD_INFO.STOPTIME, jobConstants.getJobStartTime(), jobConstants.getJobEndTime()),
-            POD_INFO.STARTTIME.greaterOrEqual(minStartTime), NODE_INFO.STARTTIME.greaterOrEqual(minStartTime)))
+            POD_INFO.STARTTIME.greaterOrEqual(jobStartOffsetDateTime),
```
Comment (sdsingh07) [inline 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java:320]: Let's add end time for pod info. If there is some delay in batch job, we are not scanning exactly 1 day data.

File: 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java [hunk 2]
```diff
@@ -355,7 +280,8 @@ private Table<? extends Record> groupByInstanceIdForEachTimeBucket(
                                        .isAlive(KUBERNETES_UTILIZATION_DATA.STARTTIME,
                                            KUBERNETES_UTILIZATION_DATA.ENDTIME, jobConstants.getJobStartTime(),
                                            jobConstants.getJobEndTime())
-                                       .and(KUBERNETES_UTILIZATION_DATA.STARTTIME.greaterOrEqual(minStartTime)))));
+                                       .and(KUBERNETES_UTILIZATION_DATA.STARTTIME.greaterOrEqual(
+                                           jobStartOffsetDateTime)))));
```
Comment (sdsingh07) [inline 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java:284]: Let's add end time for kubernetes utitlization data. If there is some delay in batch job, we are not scanning exactly 1 day data.

**Comment by abhinavsingh-ccm** [comment]
trigger uts

File: 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java (hunk 1 above)
Comment (abhinavsingh-ccm) [reply 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java:320]: Done. Ideally it should be better to create recommendation on latest data but just avoiding high scans.

File: 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java (hunk 2 above)
Comment (abhinavsingh-ccm) [reply 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/K8sRecommendationDAO.java:284]: ^^

**Review by sdsingh07 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53707#-:-NfwQRxr88Gbps7nvyWx:bnfp4nl)* status: :shipit: complete! ✓

**Comment by abhinavsingh-ccm** [comment]
trigger uts

**Pull request merged**